### PR TITLE
Bump the certgen utility to v0.2.0, and adapt the associated configuration

### PR DESCRIPTION
**Pull request by giorio94**
Bump certgen to v0.2.0, which enables the definition of the certificates to be generated via a generic configuration, agnostic of the Cilium specific details. Hence, let's refactor the certgen configuration and explicitly define the characteristics of the certificates to be generated. While being there, let's also correctly fix a few divergences with respect to the other certificate generation methods (i.e., generation of the `hubble-ui-client-certs`, and propagation of the extra DNS names and IP addresses).

Related: https://github.com/cilium/certgen/pull/220

### CODE REVIEW

File: install/kubernetes/Makefile.values [hunk 1]
```diff
@@ -25,10 +25,10 @@ else
     export HUBBLE_RELAY_REPO:=${CI_REGISTRY}/${CI_ORG}/hubble-relay-ci
 endif
 
-export CERTGEN_REPO:=quay.io/cilium/certgen
+export CERTGEN_REPO:=docker.io/giorio94/certgen
 # renovate: datasource=docker depName=quay.io/cilium/certgen
-export CERTGEN_VERSION:=v0.1.12
-export CERTGEN_DIGEST:=sha256:bbc5e65e9dc65bc6b58967fe536b7f3b54e12332908aeb0a96a36866b4372b4e
+export CERTGEN_VERSION:=v0.2.0
+export CERTGEN_DIGEST:=sha256:31dd2fccfeb3e2cf4eb47691ba197b098c71ed4ed04791854cfe8b3793047da6
```
Comment (giorio94) [inline install/kubernetes/Makefile.values]: ~~These variables still need to be appropriately updated to use the upstream certgen version once released.~~

Edit: Updated to use the upstream certgen version.

Comment (sayboras) [reply install/kubernetes/Makefile.values]: I am wondering how rennovate can update this image, do we need to run some make command after ? :thinking:

Comment (sayboras) [reply install/kubernetes/Makefile.values]: As 0.2.0 is breaking change, we probably need to double check if renovate config needs any update to restrict versions in old branches.

Comment (giorio94) [reply install/kubernetes/Makefile.values]: > I am wondering how rennovate can update this image, do we need to run some make command after ?

Yep, it seems to be configured to run it: https://github.com/cilium/cilium/blob/main/.github/renovate.json5#L18

> As 0.2.0 is breaking change, we probably need to double check if renovate config needs any update to restrict versions in old branches.

Yep, already merged: https://github.com/cilium/cilium/pull/32998

**Comment by rolinh** [comment]
> Opening for review to start gathering feedback. Marked as blocked, as it depends on [cilium/certgen#220](https://github.com/cilium/certgen/pull/220) being merged and released.

[v0.2.0](https://github.com/cilium/certgen/releases/tag/v0.2.0) of certgen is now published.

**Comment by giorio94** [comment]
> > Opening for review to start gathering feedback. Marked as blocked, as it depends on [cilium/certgen#220](https://github.com/cilium/certgen/pull/220) being merged and released.
> 
> [v0.2.0](https://github.com/cilium/certgen/releases/tag/v0.2.0) of certgen is now published.

Thanks! Switched to using the upstream image.

**Review by rolinh — CHANGES_REQUESTED** [review]
LGTM except for a minor fix for your drive-by fix for Hubble UI certs (thanks btw!).

File: install/kubernetes/cilium/templates/hubble/tls-cronjob/_job-spec.tpl [hunk 2]
```diff
@@ -19,30 +19,94 @@ spec:
...
+                  {{- range $ip := .Values.hubble.metrics.tls.server.extraIpAddresses }}
+                  - {{ $ip | quote }}
+                  {{- end }}
+                  usage:
+                  - signing
+                  - key encipherment
+                  - server auth
+                  validity: {{ $certValidityStr }}
+                {{- end }}
+                {{- if and .Values.hubble.ui.enabled .Values.hubble.relay.enabled .Values.hubble.relay.tls.server.enabled }}
```
Comment (rolinh) [inline install/kubernetes/cilium/templates/hubble/tls-cronjob/_job-spec.tpl:98]: Thanks for the drive-by fix! However, I think we only want to do that if mTLS is actually enabled on Hubble Relay's server:
```suggestion
                {{- if and .Values.hubble.ui.enabled .Values.hubble.relay.enabled .Values.hubble.relay.tls.server.enabled .Values.hubble.relay.tls.server.mtls }}
```

Comment (giorio94) [reply install/kubernetes/cilium/templates/hubble/tls-cronjob/_job-spec.tpl:98]: Hmm, actually the `mtls` value is not taken into account both in case of [helm](https://github.com/giorio94/cilium/blob/mio%2Fcertgen-config/install/kubernetes/cilium/templates/hubble/tls-helm/ui-client-certs.yaml#L1) and [cert-manager](https://github.com/giorio94/cilium/blob/mio%2Fcertgen-config/install/kubernetes/cilium/templates/hubble/tls-certmanager/ui-client-certs.yaml#L1) generation. I'd probably defer that change to a separate PR. WDYT?

**Comment by giorio94** [comment]
Rebasing to see if the Conformance Ingress and Gateway API failures get resolved.

**Review by rolinh — APPROVED** [review]

File: install/kubernetes/cilium/templates/hubble/tls-cronjob/_job-spec.tpl [hunk 3]
```diff
@@ -19,30 +19,94 @@ spec:
...
-            - "--hubble-relay-server-cert-generate"
-            - "--hubble-relay-server-cert-validity-duration={{ $certValiditySecondsStr }}"
-            {{- end }}
-            {{- if and .Values.hubble.metrics.enabled .Values.hubble.metrics.tls.enabled }}
-            - "--hubble-metrics-server-cert-generate"
-            - "--hubble-metrics-server-cert-validity-duration={{ $certValiditySecondsStr }}"
-            {{- end }}
+            - "--ca-common-name=Cilium CA"
+          env:
+            - name: CILIUM_CERTGEN_CONFIG
```
Comment (chancez) [inline install/kubernetes/cilium/templates/hubble/tls-cronjob/_job-spec.tpl:31]: Should we put this into a configMap and read it as a file instead? Might require another update to certgen though.

Comment (giorio94) [reply install/kubernetes/cilium/templates/hubble/tls-cronjob/_job-spec.tpl:31]: That would definitely be a possibility (and also supported by certgen). It seemed just a bit overkill to me in this case due to the requirement for an additional resource, which can instead be avoided with this approach, while being functionally equivalent.

Comment (giorio94) [reply install/kubernetes/cilium/templates/hubble/tls-cronjob/_job-spec.tpl:31]: Another advantage here is that the full pod manifest gets collected in sysdumps, while it would not be the case if provided through an external configmap.

Comment (chancez) [reply install/kubernetes/cilium/templates/hubble/tls-cronjob/_job-spec.tpl:31]: Fair, though we can collect the ConfigMap in the sysdump too. Not a blocker, just thought I'd check.

**Review by chancez — APPROVED** [review]

File: install/kubernetes/Makefile.values (hunk 1 above)
Comment (sayboras) [reply install/kubernetes/Makefile.values]: yeah, rennovate configuration is there,  but I am not sure if it _ever_ works. 

Anyway, it's not related to the goal of this change.

**Review by sayboras — APPROVED** [review]

**Comment by sayboras** [comment]
https://github.com/cilium/cilium/pull/33093 is merged, you might need to rebase one more time.

File: install/kubernetes/cilium/templates/hubble/tls-cronjob/_job-spec.tpl (hunk 3 above)
Comment (giorio94) [reply install/kubernetes/cilium/templates/hubble/tls-cronjob/_job-spec.tpl:31]: Sure, it just requires less changes this way.

File: install/kubernetes/Makefile.values (hunk 1 above)
Comment (giorio94) [reply install/kubernetes/Makefile.values]: I've tried running renovate locally, and it seems to correctly pick the update:

```json
{
  "depName": "quay.io/cilium/certgen",
  "currentValue": "v0.1.12",
  "currentDigest": "sha256:bbc5e65e9dc65bc6b58967fe536b7f3b54e12332908aeb0a96a36866b4372b4e",
  "datasource": "docker",
  "replaceString": "# renovate: datasource=docker depName=quay.io/cilium/certgen\nexport CERTGEN_VERSION:=v0.1.12\nexport CERTGEN_DIGEST:=sha256:bbc5e65e9dc65bc6b58967fe536b7f3b54e12332908aeb0a96a36866b4372b4e",
  "updates": [
    {
      "bucket": "latest",
      "newVersion": "v0.2.0",
      "newValue": "v0.2.0",
      "newMajor": 0,
      "newMinor": 2,
      "updateType": "minor",
      "newDigest": "sha256:169d93fd8f2f9009db3b9d5ccd37c2b753d0989e1e7cd8fe79f9160c459eef4f",
      "branchName": "renovate/all-dependencies"
    }
  ],
  "packageName": "quay.io/cilium/certgen",
  "versioning": "docker",
  "warnings": [],
  "registryUrl": "https://quay.io",
  "currentVersion": "v0.1.12",
  "isSingleVersion": true,
  "fixedVersion": "v0.1.12"
},
```
